snapshottest/module.py

```diff
@@ -254,6 +259,23 @@ def assert_match(self, value, name=''):
     def save_changes(self):
         self.module.save()
 
+    def clear_ignore_keys(self, data):
+        if isinstance(data, dict):
+            for key, value in data.items():
+                if key in self.ignore_keys:
+                    data[key] = None
+                elif isinstance(value, dict):
```

Comment:
```suggestion
                elif isinstance(value, (dict, list)):
```
This change should make it so that dicts that contain lists of dicts can be correctly ignored.

Reply:
A few lines down I already check this:
```
elif isinstance(data, list):
```

It doesn't make sense to check it on the same line since the cases must be handled differently. Also there is a added passing test `test_snapshot_assert_match__matches_with_diffing_ignore_keys` (parameterized) which tests ignoring a key of a dict inside a list.

Reply:
Hey, thanks for the quick response!

Technically you do get to this a few lines down, but there is a bit of a logic issue there for the case that I specified:

Take this example, for instance. We want to ignore the `timestamp` key:
```
{
  'should': [
    {
      'bool': {
        'must': [
          {
            'term': {
              'timestamp': 'iwanttoignorethiskey'
            }
          }
        ]
      }
    }
  ]
}
```

If your aim was to ignore this `timestamp` key, we would get into your first check of `if isinstance(data, dict):` due to `data` being a dict (the root). So, now we start iterating over the dict's keys.  In this example, the dict has a key of `should`. However, due to the `value` of `dict['should']` being a list list, the recursion stops here.
```
for key, value in data.items():
    if key in self.ignore_keys:
        data[key] = None
    elif isinstance(value, dict): # We don't continue our recursion if the value is a list
        data[key] = self.clear_ignore_keys(value)
return data # So we never continue down the chain.
```

If you implement the change I suggested, we can successfully continue down the chain and ignore the `timestamp` key.

Hopefully that helps clear things up. Thanks again for your work on this PR! Our team is looking forward to using this functionality.

Reply:
> Hey, thanks for the quick response!
> 
> Technically you do get to this a few lines down, but there is a bit of a logic issue there for the case that I specified (dict -> list -> dict):
> 
> Take this example, for instance. We want to ignore the `timestamp` key:
> 
> ```
> {
>   'should': [
>     {
>       'bool': {
>         'must': [
>           {
>             'term': {
>               'timestamp': 'iwanttoignorethiskey'
>             }
>           }
>         ]
>       }
>     }
>   ]
> }
> ```
> 
> If your aim was to ignore this `timestamp` key, we would get into your first check of `if isinstance(data, dict):` due to `data` being a dict (the root). So, now we start iterating over the dict's keys. In this example, the dict has a key of `should`. However, due to the `value` of `dict['should']` being a list, the recursion stops here.
> 
> ```
> for key, value in data.items():
>     if key in self.ignore_keys:
>         data[key] = None
>     elif isinstance(value, dict): # We don't continue our recursion if the value is a list
>         data[key] = self.clear_ignore_keys(value)
> return data # So we never continue down the chain.
> ```
> 
> If you choose to implement [the change I suggested](https://github.com/syrusakbary/snapshottest/pull/102#discussion_r366492613), we can successfully continue down the chain and ignore the `timestamp` key.
> 
> Hopefully that helps clear things up. Thanks again for your work on this PR! Our team is looking forward to using this functionality.

To add on to this @HeyHugo, I think you could actually just update the `elif` condition within the `dict` path to simply be an `else`, which would allow all of the dicts keys to run through `ignore_keys` and properly be parsed:
```
for key, value in data.items():
    if key in self.ignore_keys:
        data[key] = None
    else:
        data[key] = self.clear_ignore_keys(value)
return data
```

Reply:
Ah I see now. I didn't see there was another `isinstance()` inside there. Good catch thanks for this.

I've added tests cases that now cover this as well